### ISSUE TITLE
Is the Windows install webpage up-to-date?

### ISSUE THREAD

**Issue by info-rchitect** [Origen-SDK/origen#281]
@coreyeng I know you have been doing a lot of Windows work, is the webpage up-to-date?

thx

**Comment by chrisnappi** [comment]
I really recommend using Windows Services for Linux (https://docs.microsoft.com/en-us/windows/wsl/install-win10) if you can.  Using that you can just use the Linux install instructions.

**Comment by info-rchitect** [comment]
@chrisnappi yes that works for most folks who are creating test programs but we have users that are expanding to use Origen for analytics and they use ERB to create JMP scripts based.  Also, if anyone has any tips on getting their SSH to work on Windows 10 that would be greatly appreciated.

**Comment by chrisnappi** [comment]
@info-rchitect  - well any of the solutions (Cygwin/choco-unxutils/WSL) all end up making your Windows have Linux-like commands.  All have full file system access.  WSL is really the most seemless IMHO.

**Comment by coreyeng** [comment]
@info-rchitect  Its a few years old now. I think 2016 is when I updated it. It's probably still good for Windows 7 but I'm not sure about 10. @pderouen had said the powershell command had a syntax error, but it was easy to fix and I'm not sure if it has to do with running on powershell itself or in the plain cmd shell. We had several people install it on Windows 10 and I didn't hear of any issue, but none of them would be using powershell prompt. They all use the cmd prompt.

As for SSH, I use the [net-ssh](https://github.com/net-ssh/net-ssh) and [net-scp](https://github.com/net-ssh/net-scp) gems to copy patterns from windows back to Linux and run them through the simulator, while dumping the results back to the Windows terminal. It works pretty well, just need to watch the startup scripts. SCP in particular has issues if a new shell prints a bunch of stuff to the console.

**Comment by info-rchitect** [comment]
@coreyeng @chrisnappi  Have you seen this error when trying to install Origen on Windows 10 Ubuntu?

~~~shell
root@D10-brian:/mnt/c/origen/nemawashi# !43
gem install origen --no-rdoc --no-ri
Building native extensions.  This could take a while...
/usr/lib/ruby/2.3.0/rubygems/ext/builder.rb:76: warning: Insecure world writable dir /mnt/c in PATH, mode 040777
ERROR:  Error installing origen:
        ERROR: Failed to build gem native extension.

    current directory: /var/lib/gems/2.3.0/gems/rainbow-2.2.2/ext
/usr/bin/ruby2.3 mkrf_conf.rb
~~~

**Comment by ginty** [comment]
I don't think we are seeing the true error here, but if its failing to build a native extension it is likely that you need to install the Ubuntu build essentials first.

This should do it:

~~~text
sudo apt-get install autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm3 libgdbm-dev
~~~

I'm not sure how you installed Ruby, but I always use Rbenv which takes care of that for you I think:
https://github.com/rbenv/rbenv

**Comment by pderouen** [comment]
I helped a new user install on Windows recently. I wish I would have kept notes on the issues that we had. These are what I recall:

- The command to install chocolatey errors out. One of the commands was given the wrong number of arguments. I had him manually download and install. We are stuck on Win7. Don't know if that contributes.
- We skip installing the "better" console. I tried it once upon a time and prefer just a native PS console.
- I recall the devkit equivalent being completely different for the newer ruby release.
